drakrun/drakrun/draksetup.py

```diff
@@ -437,6 +479,24 @@ def postinstall(report, generate_usermode):
 
     install_info = InstallInfo.load()
 
+    logging.info("Cleaning up leftovers(if any)")
+    if install_info.zfs_tank_name:
+        volume_path = os.path.join("/", "dev", "zvol", install_info.zfs_tank_name, "tmp-part2")
+        try:
+            subprocess.check_output(f'umount {volume_path}', shell=True)
+            logging.info("Unmounted disk")
+        except subprocess.CalledProcessError as ext:
+            logging.exception(ext.stdout)
+
+        tmp_snap = shlex.quote(os.path.join(install_info.zfs_tank_name, "tmp"))
+        try:
+            subprocess.check_output(f"zfs destroy {tmp_snap}", shell=True)
+        except subprocess.CalledProcessError:
+            pass
```

Comment:
I don't like ZFS-specific code leaking here. Mounting and cleanup should be done in storage backend.

Reply:
Umm, this was basically a part of `postinstall` process and used in `create_rekall_profiles` and some of the job was not being done in `storage.py`
https://github.com/CERT-Polska/drakvuf-sandbox/blob/85c0371fedb4fba9c375bf248f0e0cc9c4ead6c8/drakrun/drakrun/draksetup.py#L363-L372
Here the mounting was being done which I had unmounted
Where should I move it? coz I don't think moving the unmounting part to storage backend code will be good as it is not mounted there
Should I separate this into some other function?

Reply:
I am not sure truly if it'll work but I have an idea, as we are using the `with` block I am assuming, this does the cleanup by itself if anything occurs, but apparently the file storage was mounted which led to not being able to delete the file system when exiting this block
https://github.com/CERT-Polska/drakvuf-sandbox/blob/9d755cff819a9e5ed4e3d979cce1b03b2c53ca1b/drakrun/drakrun/storage.py#L178-L180
If I add the unmounting part above this, will it work?